plugin/pkg/log/log.go

```diff
@@ -1,31 +1,36 @@
 // Package log implements a small wrapper around the std lib log package.
-// It implements log levels by prefixing the logs with [INFO], [DEBUG],
-// [WARNING] or [ERROR].
+// It implements log levels by prefixing the logs with the current time
+// with in RFC3339Milli and [INFO], [DEBUG], [WARNING] or [ERROR].
 // Debug logging is available and enabled if the *debug* plugin is used.
 //
 // log.Info("this is some logging"), will log on the Info level.
 //
-// log.Debug("this is debug output"), will log in the Debug level.
+// log.Debug("this is debug output"), will log in the Debug level, etc.
 package log
 
 import (
 	"fmt"
 	"io/ioutil"
 	golog "log"
 	"os"
+	"time"
 )
 
 // D controls whether we should output debug logs. If true, we do.
 var D bool
 
+// RFC3339Milli doesn't exist, invent it here.
+func clock() string { return time.Now().Format("2006-01-02T15:04:05.999Z07:00") }
+
 // logf calls log.Printf prefixed with level.
 func logf(level, format string, v ...interface{}) {
-	s := level + fmt.Sprintf(format, v...)
-	golog.Print(s)
+	golog.Print(clock() + " " + level + fmt.Sprintf(format, v...))
```

Comment:
minor optimizations -
1. `" " + level` - the leading space can be added to constants like `info = " [INFO] "`
2. I suppose the code `golog.Print(clock(), level, fmt.Sprintf(format, v...))` will concatenate strings more effectively

the same for `func log()` below

Reply:
done